tests/h/activity/query_test.py

```diff
@@ -143,6 +153,18 @@ def test_does_not_redirect_to_user_page_if_user_does_not_exist(self,
 
         assert check_url(pyramid_request, query) is None
 
+    def test_does_not_remove_user_term_from_query_if_user_does_not_exist(self,
+                                                                         pyramid_request,
+                                                                         unparse,
+                                                                         user_service):
+        query = MultiDict({'user': 'jose'})
+        user_service.fetch.return_value = None
+
+        check_url(pyramid_request, query, unparse=unparse)
+
+        assert query.get('user') == 'jose'
+        assert not unparse.called
```

Comment:
@robertknight @sheetaluk Why are we asserting that `unparse` wasn't called here? Doesn't the first assert meet the test that it doesn't remove the user from the query? unparse seems to be irrelevant here if I'm not mistaken

Reply:
> Doesn't the first assert meet the test that it doesn't remove the user from the query? 

Yes, indeed. @sheetaluk can you follow up on this.